TypeScript compilation errors within @libp2p/interface/src/event-target.ts

- **Version**:
```
$ npm ls | grep libp2p
├── @chainsafe/libp2p-noise@15.0.0
├── @chainsafe/libp2p-yamux@6.0.2
├── @libp2p/circuit-relay-v2@1.0.15
├── @libp2p/identify@1.0.14
├── @libp2p/mplex@10.0.15
├── @libp2p/webrtc@4.0.19
├── @libp2p/websockets@8.0.15
├── libp2p@1.2.3
```

- **Platform**:
```
$ uname -a
Linux schnippi 5.15.0-92-generic #102~20.04.1-Ubuntu SMP Mon Jan 15 13:09:14 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

- **Subsystem**:
event-target.ts

#### Severity:
Low

#### Description / Steps to reproduce the error:
```
$ tsc
node_modules/@libp2p/interface/src/event-target.ts:94:71 - error TS2345: Argument of type 'CustomEventInit<Detail>' is not assignable to parameter of type 'CustomEventInit<Detail> & EventInit & { detail: Detail; }'.
  Type 'CustomEventInit<Detail>' is not assignable to type '{ detail: Detail; }'.
    Property 'detail' is optional in type 'CustomEventInit<Detail>' but required in type '{ detail: Detail; }'.

94     return this.dispatchEvent(new CustomEvent<Detail>(type as string, detail))
                                                                         ~~~~~~

node_modules/@libp2p/interface/src/event-target.ts:111:5 - error TS2578: Unused '@ts-expect-error' directive.

111     // @ts-expect-error could be undefined
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/packages/interface/src/event-target.ts
+++ b/packages/interface/src/event-target.ts
@@ -105,22 +105,4 @@ export class TypedEventEmitter<EventMap extends Record<string, any>> extends Eve
   }
 }
 
-/**
- * CustomEvent is a standard event but it's not supported by node.
- *
- * Remove this when https://github.com/nodejs/node/issues/40678 is closed.
- *
- * Ref: https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent
- */
-class CustomEventPolyfill<T = any> extends Event {
-  /** Returns any custom data event was created with. Typically used for synthetic events. */
-  public detail: T
-
-  constructor (message: string, data?: EventInit & { detail: T }) {
-    super(message, data)
-    // @ts-expect-error could be undefined
-    this.detail = data?.detail
-  }
-}
-
-export const CustomEvent = globalThis.CustomEvent ?? CustomEventPolyfill
+export const CustomEvent = globalThis.CustomEvent

